system/include/platform.h

```diff
@@ -502,6 +481,20 @@ typedef unsigned __int64 timestamp_type;
 #endif
 #endif
 
+// get rid off warning: pass signed char to unsigned parameter
+inline bool isspace_char(char c) { return isspace((unsigned char)c) != 0; }
+inline bool isdigit_char(char c) { return isdigit((unsigned char)c) != 0; }
+inline bool isalpha_char(char c) { return isalpha((unsigned char)c) != 0; }
+inline bool isprint_char(char c) { return isprint((unsigned char)c) != 0; }
+inline bool isblank_char(char c) { return isblank((unsigned char)c) != 0; }
+inline bool isupper_char(char c) { return isupper((unsigned char)c) != 0; }
+inline bool islower_char(char c) { return islower((unsigned char)c) != 0; }
+inline bool isalnum_char(char c) { return isalnum((unsigned char)c) != 0; }
+inline bool iscntrl_char(char c) { return iscntrl((unsigned char)c) != 0; }
+inline bool isxdigit_char(char c) { return isxdigit((unsigned char)c) != 0; }
+inline int toupper_char(char c) { return toupper((unsigned char)c); }
+inline int tolower_char(char c) { return tolower((unsigned char)c); }
```

Comment:
I don't have any examples, but have you considered any unintended issues that may arise due to some subtle differences between these new definitions and the old? It appears to be functionally the same but these are so widely used there could be interesting corner cases.

1. Previously the macros were defined `if (_MSC_VER>=1400)`. Could there be any code previously using the macro that may now be using the implementation in ctype.h? Could that make a difference in any case? (I don't _think_ so...)
2. Why cast now to `(unsigned char)` instead of `(const unsigned char)` as before?
3. Why return bool now instead of int?

Reply:
All good questions. These inline functions are taken from scapps, and they have been used for +10 years.

From https://en.cppreference.com/w/cpp/string/byte/toupper:
Like all other functions from <cctype>, the behavior of std::toupper is undefined if the argument's value is neither representable as unsigned char nor equal to EOF. To use these functions safely with plain chars (or signed chars), the argument should first be converted to unsigned char.